### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by mish-sk**

### CODE REVIEW

**Comment by mish-sk** [comment]
Знімок екрана 2023-02-13 о 20 34 19
Знімок екрана 2023-02-13 о 20 34 25
Знімок екрана 2023-02-13 о 20 34 33
Знімок екрана 2023-02-13 о 20 34 47
Знімок екрана 2023-02-13 о 20 34 54

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v1

File: taxi/views.py [hunk 1]
```diff
@@ -17,3 +18,30 @@ def index(request):
...
+
+
+class CarListView(generic.ListView):
+    model = Car
+    queryset = Car.objects.select_related("manufacturer").order_by("model")
+    paginate_by = 5
+
+
+class CarDetailView(generic.DetailView):
+    model = Car
```
Comment (ArsenPidhoretskyi) [inline taxi/views.py:37]: {% for driver in car.drivers.all %}
In your template. You have n + 1 problem for now

File: taxi/views.py [hunk 2]
```diff
@@ -17,3 +18,30 @@ def index(request):
...
+
+
+class DriverListView(generic.ListView):
+    model = Driver
+    paginate_by = 5
+
+
+class DriverDetailView(generic.DetailView):
+    model = Driver
+    queryset = Driver.objects.all().prefetch_related("cars")
```
Comment (ArsenPidhoretskyi) [inline taxi/views.py:47]: Do not forgot about manufacturer also

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: taxi/views.py (hunk 1 above)
Comment (ArsenPidhoretskyi) [reply taxi/views.py:37]: Not fixed need to prefetch drivers

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]
Please Update requirements. Checks are failed

**Review by ArsenPidhoretskyi — APPROVED** [review]
Well done!

**Pull request closed without merging**